src/matlab2tikz.m

```diff
@@ -4005,22 +4010,54 @@ function warnFacetedInterp(m2t)
     else
         arrowLength = '{sqrt((\thisrow{u})^2+(\thisrow{v})^2)}';
     end
+
     plotOpts = opts_add(plotOpts, 'point meta', arrowLength);
+    plotOpts = opts_add(plotOpts, 'point meta min', '0');
 
     if showArrowHead
-        %TODO: scale the arrows more rigorously to match MATLAB behavior
-        %There is a "MaxHeadSize" property (at least in R2014b) that plays a
-        %role in determining the quiver size.
         arrowHeadOpts = opts_new();
-        arrowHeadOpts = opts_add(arrowHeadOpts, 'scale length', ...
-                                 '{max(0.01,\pgfplotspointmetatransformed/1000)}');
-        arrowHeadOpts = opts_add(arrowHeadOpts, 'scale width', ...
-                                 '{0.5*max(0.01,\pgfplotspointmetatransformed/1000)}');
+
+        % In MATLAB (HG1), the arrow head is constructed to have an angle of
+        % approximately 18.263 degrees in 2D as can be derived from the
+        % |quiver| function.
+        % In 3D, the angle is no longer constant but it is approximately
+        % the same as for 2D quiver plots. So let's make our life easy.
+        % |test/examples/example_quivers.m| covers the calculations.
+        arrowHeadOpts = opts_add(arrowHeadOpts, 'angle''', '18.263');
+
+        % TODO: scale the arrows more rigorously to match MATLAB behavior
+        % Currently, this is quite hard to do, since the size of the arrows
+        % is defined in pgfplots in absolute units (here we specify that those
+        % should be scaled up/down according to the data), while the data itself
+        % is in axis coordinates (or some scaled variant). I.e. we need the
+        % physical dimensions of the axis to compute the correct scaling!
+        % 
+        % There is a "MaxHeadSize" property that plays a role.
+        % MaxHeadSize is said to be relative to the length of the quiver in the
+        % MATLAB documentation. However, in practice, there seems to be a SQRT
+        % involved somewhere (e.g. if u.^2 + v.^2 == 2, all MHS values >
+        % 1/sqrt(2) are capped to 1/sqrt(2)).
```

Comment:
It's a bit long ago for me, you may be very right on the intent, but the way is a bit weird: note that it seems to be capped to 1/sqrt(L) where L is the length of the quiver. Anyhow, in this case I would say that the intent is one thing, re-implementing the behavior is another thing (since I'm not convinced on the relationship).